robber/matchers/exception.py

```diff
@@ -0,0 +1,33 @@
+from robber import expect
+from robber.matchers.base import Base
+
+
+class ExceptionMatcher(Base):
+    """
+    expect(lambda: call_something(with_some_params)).to.throw(any_exception)
+    """
+    @property
+    def _raised(self):
+        raised = None
+
+        if not callable(self.actual):
+            raise TypeError('{actual} is not callable'.format(actual=self.actual))
+
+        try:
+            self.actual()
+        except BaseException as raised:
+            return raised
+
+    def matches(self):
+        return isinstance(self._raised, self.expected)
```

Comment:
I think we should only use `isinstance` when we don't need to care about `super` classes. But when we do, how about `type() ==`?